feat(select): remove role from action sheet interface (#31221)
driftyco/ionic: BREAKING.md

Reviewer: Should we add a header above this now since there are two sections?

Author: https://github.com/ionic-team/ionic-framework/pull/31221/changes/376fef9b72af478a005110039b18a76785322b85

Apps that relied on `ionChange` firing on every confirmation (for example, to detect overlay dismissal without a value change) should listen for `ionDismiss` instead, or use the `didDismiss` event on the underlying alert or action sheet.

**Action Sheet Interface `selected` Role Removed**

> When using `interface="action-sheet"`, `ion-select` no longer assigns the `selected` role to the action sheet button for the currently selected option. This aligns the `action-sheet` interface with the `alert`, `popover`, and `modal` interfaces, none of which assign this role. The selected option's styling is unchanged because it is driven by the option's checked state.

Reviewer: ```suggestion
When using `interface="action-sheet"`, `ion-select` no longer assigns the `selected` role to the action sheet button for the currently selected option. This aligns the `action-sheet` interface with the `alert`, `popover`, and `modal` interfaces, none of which assign this role. This does not change the selected option's styling.
```

I think this might be giving too much information. We don't need to explain what controls the styling.

Author: https://github.com/ionic-team/ionic-framework/pull/31221/changes/9962d4023afe36608f7e20618af79c5ddedf2463

> Because the option button no longer carries the `selected` role, the action sheet's dismiss role for a selected option is no longer `"selected"`. Apps that inspected this role to detect that a value was chosen, such as reading `role` from the underlying action sheet's `onDidDismiss` result, should listen for `ion-select`'s `ionChange` event instead, which emits the selected value when the selection changes.

Reviewer: When was the `"selected"` role being output? Even on `main` it just shows as `''`:

<img width="740" height="106" alt="Image" src="https://github.com/user-attachments/assets/df6b9b3f-7481-45a6-a4ce-fc24b15112a7" />

Author: This would happen only when selecting the same value. Exact steps can be found in the Jira ticket. I updated the Breaking doc to clarify it: https://github.com/ionic-team/ionic-framework/pull/31221/changes/e546e913bd42fedc0289169d748fd91bdbd31d73

Reviewer: Ahh okay. Shouldn't it just say the role is now `undefined` in both cases?

Author: Better? https://github.com/ionic-team/ionic-framework/pull/31221/changes/a53543faf304d79a8a354f972abc38ccee01c305

Reviewer: Yeah looks good!
